Tools/jit/_stencils.py

```diff
@@ -209,7 +209,15 @@ def pad(self, alignment: int) -> None:
             self.disassembly.append(f"{offset:x}: {' '.join(['00'] * padding)}")
         self.body.extend([0] * padding)
 
-    def remove_jump(self, *, alignment: int = 1) -> None:
+    def add_nop(self, alignment: int) -> None:
+        """Add a NOP if the offset is not aligned."""
+        offset = len(self.body)
+        nop = b"\x1f\x20\x03\xD5"
+        if offset % alignment:
+            self.disassembly.append(f"{offset:x}: d503201f\t\t nop")
+            self.body.extend(nop)
```

Comment:
I don't like that this is ARM-specific, and only really happens to work because of the current alignment values we chose.

I wonder how messy it would be to make `nop` an attribute of `_Target`, and pass it into `process_relocations`, which passes it here (sort of like `alignment`).

I think that we should probably also make this so that it can add several `nop`s if the alignment is larger (such as 16), and fail if this isn't possible (when adding `nops` overshoots the intended alignment).

Reply:
Have a look. I think I have addressed all your comments.